Makefile

```diff
@@ -1,44 +1,62 @@
 .PHONY: test test-coverage test-coverage-html test-coverage-clover test-unit test-unit-coverage test-unit-coverage-html test-unit-coverage-clover test-functional test-functional-coverage test-functional-coverage-html test-functional-coverage-clover install
 
-# tests
-test:
-	@./vendor/bin/phpunit
+# Setting up
 
-test-coverage:
-	@./vendor/bin/phpunit --coverage-text
+setup: ## Install dependencies and set up example conf file
+	@docker-compose run --rm composer install
+
+# Testing
+
+test: ## Run all tests
+test: test-coverage test-unit test-unit-coverage test-functional test-functional-coverage
+
+# Coverage tests
+test-coverage: ## Run coverage tests
+	@docker-compose run --rm php-55 ./vendor/bin/phpunit --coverage-text
 
 test-coverage-html:
-	@./vendor/bin/phpunit --coverage-html ./tests/report/html
+	@docker-compose run --rm php-55 ./vendor/bin/phpunit --coverage-html ./tests/report/html
 
 test-coverage-clover:
-	@./vendor/bin/phpunit --coverage-clover=./tests/report/coverage.clover
+	@docker-compose run --rm php-55 ./vendor/bin/phpunit --coverage-clover=./tests/report/coverage.clover
 
-# unit tests
-test-unit:
-	@./vendor/bin/phpunit --testsuite unit
+# Unit tests
+test-unit: ## Run unit tests
+	@docker-compose run --rm php-55 ./vendor/bin/phpunit --testsuite unit
 
 test-unit-coverage:
-	@./vendor/bin/phpunit --testsuite unit --coverage-text
+	@docker-compose run --rm php-55 ./vendor/bin/phpunit --testsuite unit --coverage-text
```

Comment:
I know these were already here, but I don't think we need separate coverage targets for unit and functional tests. The purpose of reporting on coverage is to see how much of the code overall is covered by tests. I think there is little value in seeing a subset of the coverage.

I think you can get rid of all the coverage targets except these:

* `test-coverage`
* `test-coverage-html`
* `test-coverage-clover`